addons/website_sale_slides/models/product_product.py

```diff
@@ -9,6 +9,15 @@ class Product(models.Model):
 
     channel_ids = fields.One2many('slide.channel', 'product_id', string='Courses')
 
+    def _is_add_to_cart_allowed(self):
+        # Allow course related products to the cart regardless of product's rules
+        self.ensure_one()
+        res = super()._is_add_to_cart_allowed()
+        return res or self.env['slide.channel'].sudo().search_count([
+            ('is_published', '=', True),
```

Comment:
I think it's better to use "website_published", as the courses model use the "website.published.multi.mixin" and the "website_published" field search method takes into account the current website in addition to the fact that it's published or not.